library/lists.pl

```diff
@@ -552,6 +553,21 @@
     Sum1 is Sum0 + X,
     sum_list(Xs, Sum1, Sum).
 
+%!  prod_list(+List, -Sum) is det.
+%
+%   Prod is the result of multiplying all numbers in List.
```

Comment:
You should specify the result if List is [].

In essence, you're defining:
```
prod_list(List, Prod) :- foldl(times, List, Prod).
times(X, Y, Prod) :- Prod is X * Y.
```

Reply:
Yes, the predicate definition is equivalent. However, when a 0 is present in list, we can stop multiplying all the numbers and return 0. This is not easy to implement with `foldl/N`. Moreover, for consistency, my definition is similar to the one of `sum_list/2` (which can as well be implemented as `foldl(my_sum,L,0,R1)` with `my_sum(A,B,C):- C is A + B.`)